Library/Homebrew/cmd/reinstall.rb

```diff
@@ -2,6 +2,14 @@
 
 module Homebrew
   def reinstall
+    if !MacOS.can_build?
```

Comment:
I'm not sure which logic you're referring to in `FormulaInstaller#install` - the bottle checking? If so, I think that's possible.